File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/domain/like/service/WriteLikeService.java
package io.dustin.apps.board.domain.like.service;

import io.dustin.apps.board.domain.like.model.Like;
import io.dustin.apps.board.domain.like.repository.LikeRepository;
import io.dustin.apps.common.code.BoardType;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class WriteLikeService {

    private final LikeRepository likeRepository;

    public Like create(Long userId, Long boardId, BoardType boardType){
        Like like = Like.builder()
                .userId(userId)
                .boardId(boardId)
                .boardType(boardType)
                .build();
        this.likeRepository.save(like);
        return like;
    }

    @Transactional
    public void delete(Long userId, Long boardId, BoardType boardType){
        this.likeRepository.deleteByUserIdAndBoardIdAndBoardType(userId, boardId, boardType);
    }
}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/like/service/WriteLikeService.kt
package io.dustin.apps.board.domain.like.service

import io.dustin.apps.board.domain.like.model.Like
import io.dustin.apps.board.domain.like.repository.LikeRepository
import io.dustin.apps.common.code.BoardType
import org.springframework.stereotype.Service

@Service
class WriteLikeService (
    val likeRepository: LikeRepository
) {

    /**
     * 좋아요
     */
    fun create(userId: Long, boardId: Long, boardType: BoardType): Like {
        val like = Like(
            userId = userId,
            boardId = boardId,
            boardType = boardType
        )
        likeRepository.save(like)
        return like
    }

    /**
     * 좋아요 취소
     */
    fun delete(userId: Long, boardId: Long, boardType: BoardType) {
        likeRepository.deleteByUsers(userId, boardId, boardType)
    }
}
